crates/red_knot_project/src/lib.rs

```diff
@@ -626,10 +627,27 @@ where
             ));
 
             if let Some(backtrace) = error.backtrace {
-                diagnostic.sub(SubDiagnostic::new(
-                    Severity::Info,
-                    format!("Backtrace:\n{backtrace}"),
-                ));
+                match backtrace.status() {
+                    BacktraceStatus::Disabled => {
+                        diagnostic.sub(SubDiagnostic::new(
+                            Severity::Info,
+                            "run with `RUST_BACKTRACE=1` environment variable to display a backtrace",
```

Comment:
That message might be slightly confusing to users in the presence of the query backtrace that we seem to show unconditionally (right below this message).

Maybe
```suggestion
                            "run with `RUST_BACKTRACE=1` environment variable to show the full backtrace information",
```